File: exercises/testing/find-the-bugs-weather/app/src/main/java/ch/epfl/sweng/project/geocoding/GeocodingService.java
package ch.epfl.sweng.project.geocoding;

import androidx.annotation.NonNull;

import java.io.IOException;

import ch.epfl.sweng.project.location.Location;

/**
 * This service enables you to get a {@link ch.epfl.sweng.project.location.Location} from a String
 * representing a location (i.e. a city, a street, a country...), or to get a {@link Address} from
 * a {@link ch.epfl.sweng.project.location.Location}
 */
public interface GeocodingService {

    /**
     * Gets the location of a textual address
     *
     * @param address the address, i.e. Lausanne, Switzerland
     * @return the location for this address
     *
     * @throws IOException if the network is unavailable
     */
    Location getLocation(@NonNull String address) throws IOException;

    /**
     * Get the address corresponding to a location
     *
     * @param location the location for which you want an address
     * @return the address of this location
     *
     * @throws IOException if the network is unavailable
     */
    Address getAddress(@NonNull Location location) throws IOException;
}

File: exercises/devops/sample-app-without-ci/app/src/main/java/ch/epfl/sweng/project/geocoding/GeocodingService.kt
package ch.epfl.sweng.project.geocoding

import ch.epfl.sweng.project.location.Location
import java.io.IOException

/**
 * This service enables you to get a [ch.epfl.sweng.project.location.Location] from a String
 * representing a location (i.e. a city, a street, a country...), or to get a [Address] from
 * a [ch.epfl.sweng.project.location.Location]
 */
interface GeocodingService {
    /**
     * Gets the location of a textual address
     *
     * @param address the address, i.e. Lausanne, Switzerland
     * @return the location for this address
     *
     * @throws IOException if the network is unavailable
     */
    @Throws(IOException::class)
    fun getLocation(address: String): Location?

    /**
     * Get the address corresponding to a location
     *
     * @param location the location for which you want an address
     * @return the address of this location
     *
     * @throws IOException if the network is unavailable
     */
    @Throws(IOException::class)
    fun getAddress(location: Location): Address?
}